pkg/lib/machines.js

```diff
@@ -176,28 +172,23 @@
         }
 
         function update_saved_machine(host, values) {
-            function mutate(data) {
-                if (!data)
-                    data = { };
-                var item = data[host];
-                if (!item)
-                    item = data[host] = { };
-                merge(item, values);
-                return data;
-            }
-
-            /* Update the JSON file */
-            var local = cockpit.file(path, { syntax: JSON, superuser: "try" });
-            var mod = local.modify(mutate, last.content, last.tag)
-                .done(function(data, tag) {
+            // wrap values in variants for D-Bus call
+            var values_variant = {};
+            for (var prop in values)
+                if (values[prop] !== null)
+                    values_variant[prop] = cockpit.variant(prop == "visible" ? 'b' : 's', values[prop]);
+
+            // FIXME: investigate re-using the proxy from Loader (runs in different frame/scope)
+            var bridge = cockpit.dbus(null, { bus: "internal", "superuser": "try" });
+            var mod = bridge.call("/machines", "cockpit.Machines", "Update", [ "99-webui.json", host, values_variant ])
```

Comment:
That's a pretty good reason. And I think one that we can live with. 

A bigger decision is the implementation of multiple machines per file, rather than enforcing one machine per file. However I think you've made the right call here too.